tests/test_cli_help_env.py

```
@@ -56,21 +57,47 @@ def test_completions_emits_script(state_env) -> None:
+    assert "#compdef polylogue" in result.output
+    assert "compdef _polylogue_complete polylogue" in result.output
+
+
 def test_fish_completions_include_descriptions(state_env) -> None:
     runner = CliRunner()
     result = runner.invoke(click_cli, ["completions", "--shell", "fish"])
     assert result.exit_code == 0
     assert "Synchronize provider archives" in result.output
 
 
+def test_completions_auto_detects_shell_from_env(state_env) -> None:
```

Ruff (0.15.9)

[warning] Unused function argument: `state_env`

(ARG001)